linux/device.go

```diff
@@ -870,7 +870,7 @@ func CreateNimbleDevices(vols []*model.Volume) (devs []*model.Device, err error)
 	var devices []*model.Device
 	for _, vol := range vols {
 		log.Tracef("create request with serialnumber :%s, accessprotocol %s discoveryip %s, iqn %s ", vol.SerialNumber, vol.AccessProtocol, vol.DiscoveryIP, vol.Iqn)
-		if vol.AccessProtocol == iscsi && (vol.DiscoveryIP == "" || vol.Iqn == "") {
+		if vol.AccessProtocol == iscsi && (vol.DiscoveryIP == "" || vol.Iqn == "") && len(vol.DiscoveryIPs) == 0 {
```

Comment:
i think this should be || instead of &&, as older versions doesn't set DiscoveryIP's, but just DiscoveryIP

Reply:
since this is error condition check which means neither of discoveryIp or discoveryIps are set that is why && condition if any of them is set we should be fine.